### DISCUSSION TITLE
[Feature]: Select Julia version at first import.

### DISCUSSION THREAD

**Discussion by JacobZhao** [MilesCranmer/PySR#576] [Q&A] [answered]
### Feature Request

It would be better if user can choose correct julia version during first time import.
For instance, i running Pysr on Macbook Air M2, during first time import. Programm automatically select julialang for x64 version instead of aarch64. 
Eventhough program successfully installed, but it faceing the translation loss from AMD64 to Apple64.

**Comment by MilesCranmer** [comment, marked as answer]
It only downloads Julia if it can't already find it on your system. You can provide the location as follows (see JuliaCall docs here: https://juliapy.github.io/PythonCall.jl/stable/juliacall/#julia-config)

```python
import os
os.environ["PYTHON_JULIACALL_BINDIR"] = "/path/to/julia/binary/folder"
from pysr import PySRRegressor
```

or by starting Python like this:

```bash
python -X juliacall-home='/path/to/julia/binary/folder' script.py
```

Just note that it will still re-download Julia if you have an incompatible version. Right now PySR skips Julia 1.10.1 and 1.10.2 due to a bug in libgomp. So if you had one of those versions, you would need to downgrade to 1.10.0 to avoid redownloading.

**Comment by MilesCranmer** [comment]
Also, if possible, please submit a bug report to PythonCall.jl: https://github.com/JuliaPy/PythonCall.jl describing your setup. Because it should have downloaded the aarch64 version rather than the x64 one; that seems like a bug.

**Reply by JacobZhao** [reply]
Hi Miles, i try to rebulit Pysr in a new venv, this time it works, every thing is fine.
Best,
Zhao
